packages/discussions/src/utils/channel-permission.ts

```diff
@@ -0,0 +1,262 @@
+import { IGroup } from "@esri/arcgis-rest-types";
+import {
+  AclCategory,
+  AclSubCategory,
+  IChannelAclPermission,
+  IChannelAclPermissionDefinition,
+  IDiscussionsUser,
+  Role,
+} from "../types";
+import { CANNOT_DISCUSS } from "./constants";
+import { isOrgAdmin } from "./platform";
+
+type PermissionsByAclCategoryMap = {
+  [key in AclCategory]?: IChannelAclPermission[];
+};
+
+export class ChannelPermission {
+  private readonly ALLOWED_GROUP_MEMBER_TYPES = ["owner", "admin", "member"];
+  private readonly ALLOWED_ROLES_FOR_POSTING = [
+    Role.WRITE,
+    Role.READWRITE,
+    Role.MANAGE,
+    Role.MODERATE,
+    Role.OWNER,
+  ];
```

Comment:
I think there is a way to convert the `Role` enum to an array without manually enumerating.

Reply:
👍